z3c/dependencychecker/modules.py

```diff
@@ -0,0 +1,398 @@
+# -*- coding: utf-8 -*-
+from xml.etree import ElementTree
+from z3c.dependencychecker.dotted_name import DottedName
+import ast
+import fnmatch
+import os
+import re
+
+
+TEST_REGEX = r"""
+{0}    # Needs to start with the path separator
+\w*    # There could be some characters (or none)
+test   # has to contain test on it
+s?     # ensure plurals are handled as well
+(      # Needs to end with either a path or a file extension:
+{0}    # path separator
+|      # OR
+\w*    # some possible characters
+\.     # a dot
+\w+    # extension
+)
+""".format(os.sep)
+
+TEST_IN_PATH_RE = re.compile(
+    TEST_REGEX,
+    re.VERBOSE,
+)
+
+
+class BaseModule(object):
+
+    def __init__(self, package_path, full_path):
```

Comment:
I see, sorry for the confusion. Let me tell its story :bonfire:

Up until a week or so there was only a single argument, ``path`` stored in ``self.path``.

But then while working on the regex I noticed that as I was passing the full path (i.e. self.path) to the file, if the git checkout of the package being analyzed happened to be on ``/home/gil/source/tests/my.package``, then every single module analyzed would be marked as a test module.

For that I introduced a second argument on the init method (package_path) so that I could get only the relevant part of the filesystem path and thus avoid those problems.

If you look at the usage of ``_relative_path`` is only being used in ``_is_test_module`` method, nowhere else.

That's why I decided to keep it as a private variable and keep the other variable as ``path``.

Does that make more sense now? Probably a docstring on ``__init__`` explaining that would help?